tests/test_models.py

```diff
@@ -2525,6 +2525,31 @@ def test_replace_contributor(self):
             contrib.unclaimed_records.keys()
         )
 
+    def test_permission_update_on_readded_contributor(self):
+        # A user is added as a contributor
```

Comment:
I think this test is testing slightly different behavior (if a user gets added to a project with the correct permissions). Can you modify it to check that a contributor's permissions for a node get overridden when they are added from the parent project? Something like...
1) Add user to node with certain permissions
2) Add user to parent project with different permissions 
3) Assert that user node permissions are now the same as on the parent project.